File: fl_amap_map/ios/Classes/map/AMapPlatformViewFactory.swift
import Flutter

public class AMapPlatformViewFactory: NSObject, FlutterPlatformViewFactory {
    private var registrar: FlutterPluginRegistrar

    init(_ registrar: FlutterPluginRegistrar) {
        self.registrar = registrar
        super.init()
    }

    public func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        let channel = FlutterMethodChannel(name: "fl_amap_map_\(viewId)", binaryMessenger: registrar.messenger())
        let options = args as! [String: Any]
        return AMapView(channel: channel, frame: frame, viewId: viewId, options: options)
    }

    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        FlutterStandardMessageCodec.sharedInstance()
    }

}

File: fl_amap_map/android/src/main/kotlin/fl/amap/map/map/AMapPlatformViewFactory.kt
package fl.amap.map.map

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class AMapPlatformViewFactory(
    private val binaryMessenger: BinaryMessenger,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        val channel = MethodChannel(binaryMessenger, "fl_amap_map_$viewId")
        return AMapView(context, channel, viewId, args as Map<*, *>)
    }

}
